### PR TITLE
#3393 - Add new keyboard shortcut "ctrl+alt+V" for pasting as SMARTS

### PR DESCRIPTION
**Pull request by AKZhuk**
## How the feature works? / How did you fix the issue?

Backend ticket https://github.com/epam/Indigo/issues/1303

Added two new options in layout "options":

input-format - set input format to load from. same values as used in output-format
output-content-type - if set to 'application/json' - functions will return json with next format:
{
"struct": output_molecule,
"format": output_format,
"original_format": original_format,
}
Where 'output_molecule' - molecule in output format, 'output_format' - output format name like 'chemical/x-daylight-smiles', 'original_format' - format used to load molecule.

## Check list
- [ ] unit-tests written
- [ ] e2e-tests written
- [ ] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [x] base branch (master or release/xx) is correct
- [x] task status changed to "Code review"
- [x] reviewers are notified about the pull request

Tickets  for failed tests:
https://github.com/epam/Indigo/issues/1355
https://github.com/epam/Indigo/issues/1357
https://github.com/epam/Indigo/issues/1360

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -13398,9 +13398,9 @@
       }
     },
     "node_modules/indigo-ketcher": {
-      "version": "1.14.0-rc.6",
-      "resolved": "https://registry.npmjs.org/indigo-ketcher/-/indigo-ketcher-1.14.0-rc.6.tgz",
-      "integrity": "sha512-G+OxAQy9SGSh8ww4AZaw0xD6J5+N+w+nxsXO9be4tvGXzAV54GFlPDGUkKgLbPZBFeGQMqUZux1YfORlJRfPVw=="
+      "version": "1.15.0-dev.4",
```
[the remark this answers is not in the document]
Comment (AKZhuk) [reply package-lock.json:13401]: Yes, for this feature we need a new Indigo  version

**Review by AnastasiyaPiatrovaKlu — APPROVED** [review]

**Comment by urszula-gawlowska** [comment]
Previewed. There are no comments.

**Review by urszula-gawlowska — APPROVED** [review]

File: packages/ketcher-standalone/src/infrastructure/services/struct/indigoWorker.ts [hunk 2]
```diff
@@ -109,8 +109,15 @@ self.onmessage = (e: MessageEvent<InputMessage<CommandData>>) => {
...
       handle(
-        (indigo, indigoOptions) =>
-          indigo.layout(data.struct, data.format, indigoOptions),
+        (indigo, indigoOptions) => {
+          const response = indigo.layout(
+            data.struct,
+            data.format,
+            indigoOptions,
+          );
+          const { struct } = JSON.parse(response);
```
Comment (Nitvex) [inline packages/ketcher-standalone/src/infrastructure/services/struct/indigoWorker.ts:118]: Why have you decided to omit other values from layout? I find it very risky

File: packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts [hunk 3]
```diff
@@ -315,6 +317,7 @@ class IndigoService implements StructService {
       const commandOptions: CommandOptions = {
         ...this.defaultOptions,
         ...options,
+        'output-content-type': 'application/json',
```
Comment (Nitvex) [inline packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts:320]: I suggest provide detailed description as per PR template. 
My main concern here is that you're changing the logic, which worked before w/o explaining what was changed

File: packages/ketcher-react/src/script/ui/component/cliparea/cliparea.jsx [hunk 4]
```diff
@@ -73,6 +74,23 @@ class ClipArea extends Component {
           event.preventDefault();
         }
       },
+      keydown: async (event) => {
+        if (this.props.focused() && this.props.onPaste) {
```
Comment (Nitvex) [inline packages/ketcher-react/src/script/ui/component/cliparea/cliparea.jsx:78]: I propose to remove nesting. The simplest way:
```
if (!this.props.focused() || !this.props.onPaste)) {
 return
}

if (...)
```

File: packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts (hunk 3 above)
Comment (AKZhuk) [reply packages/ketcher-standalone/src/infrastructure/services/struct/standaloneStructService.ts:320]: done

File: packages/ketcher-standalone/src/infrastructure/services/struct/indigoWorker.ts (hunk 2 above)
Comment (AKZhuk) [reply packages/ketcher-standalone/src/infrastructure/services/struct/indigoWorker.ts:118]: Previously layout return just structure.
Now we set output-content-type = 'application/json'  in payload options. 
 If set to 'application/json' - function will return json with next format:
{
"struct": output_molecule,
"format": output_format,
"original_format": original_format,
}
So I believe it is expected.

File: packages/ketcher-react/src/script/ui/component/cliparea/cliparea.jsx (hunk 4 above)
Comment (AKZhuk) [reply packages/ketcher-react/src/script/ui/component/cliparea/cliparea.jsx:78]: done

File: packages/ketcher-react/src/script/ui/component/cliparea/cliparea.jsx [hunk 5]
```diff
@@ -148,6 +166,19 @@ function paste(cb, formats) {
   return data;
 }
 
+async function pasteByKeydown(cb) {
```
[the remark this answers is not in the document]
Comment (AKZhuk) [reply packages/ketcher-react/src/script/ui/component/cliparea/cliparea.jsx:169]: done

**Pull request merged**